sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -647,8 +647,15 @@ def __init__(
     _validate_constructor_args_hf_pipeline(self._task, self._model)
 
   def _deduplicate_device_value(self, device: str):
+    current_device = device.upper() if device else None
     if 'device' not in self._load_pipeline_args:
-      if device == 'CPU':
+      if (not current_device and current_device != 'CPU' and
```

Comment:
This should be `if current_device and ...` (remove the `not`). As it is, this will throw in the default case.

Also, we should move this to the top level (outside of the `if 'device' not in self._load_pipeline_args:`)

Reply:
urgh.. missed that..

Reply:
we could move that outside as well but that'll be overridden anyway if load_pipeline_args has device which is why I kept it here initially..

Reply:
That's fine if we're throwing on that branch anyways